packages/kit/src/core/postbuild/analyse.js

```diff
@@ -151,20 +108,88 @@ async function analyse({ manifest_path, env }) {
 	return metadata;
 }
 
+/**
+ * @param {import('types').SSRRoute} route
+ * @param {import('types').SSREndpoint} mod
+ */
+function analyse_endpoint(route, mod) {
+	/** @type {import('types').PrerenderOption} */
+	let prerender = false;
+
+	if (mod.prerender !== undefined) {
+		validate_server_exports(mod, route.id);
```

Comment:
This should always be called, right? Not just if we're prerendering

Reply:
good catch, thanks — actually realised we can simplify that whole thing a bit